nipype/interfaces/base/tests/test_specs.py

```diff
@@ -420,18 +420,17 @@ def test_ImageFile():
     # setup traits
     x.add_trait('nifti', nib.ImageFile(types=['nifti1', 'dicom']))
     x.add_trait('anytype', nib.ImageFile())
-    x.add_trait('newtype', nib.ImageFile(types=['nifti10']))
+    with pytest.raises(ValueError):
+        x.add_trait('newtype', nib.ImageFile(types=['nifti10']))
     x.add_trait('nocompress',
                 nib.ImageFile(types=['mgh'], allow_compressed=False))
 
     with pytest.raises(nib.TraitError):
         x.nifti = 'test.mgz'
     x.nifti = 'test.nii'
     x.anytype = 'test.xml'
-    with pytest.raises(AttributeError):
-        x.newtype = 'test.nii'
```

Comment:
Should this not still produce an `AttributeError`?

Reply:
This was a bad test - it has been replaced by lines 423-424. When defining an `Image File` with an invalid `types` metadata should have raised a `ValueError`. Now it does.

In other words `x.newtype` is not created (nor added) because of the bad value passed to `types`.